python/lsst/daf/butler/registry/_registry.py

```diff
@@ -164,6 +169,15 @@ def __init__(self, database: Database, universe: DimensionUniverse, *,
                                                                  opaque=self._opaque,
                                                                  datasets=datasets,
                                                                  universe=self.dimensions)
+            context.addInitializer(lambda db: versions.storeVersions(self._attributes))
+
+        # This call does not do anything right now as we do not have a way to
+        # split tables between sub-schemas yet.
```

Comment:
When we do, I assume the goal will be to do the digest only on the "static" tables, not the "dynamic" tables whose presence and definition depends on e.g. dataset type?  That seems much easier to implement, even though there would be a range of schema changes the digest checks would not catch (i.e. those that affect only dynamic tables).

The catch is that we treat all datastores more like "dynamic" tables (they're created by calls to `self._opaque`, and even when making a new repo that won't happen by this point in the code), so that may make it hard to create digests for those.

Reply:
Yes, dynamic tables cannot be in digest as they can be made much later in the lifetime of the schema. For those tables I think we can only rely on developers to increment version numbers without help from digest.